### PR TITLE
Remove keys with null-values from ErrorNormalizer

### PR DESCRIPTION
**Pull request by mariusbalcytis**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -15,14 +15,14 @@
         }
     },
     "require": {
-        "php": ">=7.0",
+        "php": ">=7.0,<7.4",
```
Comment (mariusbalcytis) [inline composer.json:18]: Tests do not work with 7.4, travis also does not include that version – I think it's best to make it work when we'd need that.

File: src/Normalizer/ErrorNormalizer.php [hunk 2]
```diff
@@ -17,6 +17,7 @@ class ErrorNormalizer implements NormalizerInterface, TypeAwareInterface
      */
     public function normalize($result, NormalizationContext $normalizationContext)
     {
+        $normalizationContext->markNullValuesForRemoval();
```
Comment (vbartusevicius) [inline src/Normalizer/ErrorNormalizer.php:20]: Don't see the actual need for this, apart from being more aesthetic.

This way you loose fixed structure, in other languages, you'll have to check additionally for whether or not a property is defined. Of course, you may argue that in JS you can use `optional chaining` (which is still proposal) or even use `lodash`:
```javascript
error?.errors?.map(error => <ErrorBlock error={error} />);
get(error, 'errors', []).map(...);
```
But I think this adds more cognitive load when compared to plain:
```javascript
error.errors.map(error => <ErrorBlock error={error} />);
```

I would suggest even to change the line `27` to always return an array or having this option configurable.

Comment (mariusbalcytis) [reply src/Normalizer/ErrorNormalizer.php:20]: Not returning `null` where `array` is expected is actually the main point here. But always returning `errors` even with simple error codes seems a bit strange, too.. For example:
```
{"error": "limit_violation","errors":[]}
```

This field is only used for validation-based errors with translated items for each of the fields.

Comment (vbartusevicius) [reply src/Normalizer/ErrorNormalizer.php:20]: You look from the resource perspective, I see it from the programmer's.
It's easier when you don't have to check for mixed types: check whether it's defined, then check if it's an array, only then iterate over it.

**Pull request merged**